python/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
+
+include(../fetch_rapids.cmake)
+
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-find)
+
+set(CUML_VERSION 22.08.00)
+
+project(
+  cuml-python
+  VERSION ${CUML_VERSION}
+  LANGUAGES # TODO: Building Python extension modules via the python_extension_module requires the C
+            # language to be enabled here. The test project that is built in scikit-build to verify
+            # various linking options for the python library is hardcoded to build with C, so until
+            # that is fixed we need to keep C.
+            C CXX
+)
+
+################################################################################
+# - User Options  --------------------------------------------------------------
+option(FIND_CUML_CPP "Search for existing CUML C++ installations before defaulting to local files" OFF)
+option(SINGLEGPU "Disable all mnmg components and comms libraries" OFF)
+
+message(VERBOSE "CUML_PY: Searching for existing CUML C++ installations before defaulting to local files: ${FIND_CUML_CPP}")
```

Comment:
FWIW you can use the `CMAKE_MESSAGE_CONTEXT` variable to help set prefixes like `CUML_PY` for all messages within a scope.

Reply:
#4843